dmaws/build.py

```diff
@@ -41,6 +44,50 @@ def get_current_ref(cwd):
     return run_git_cmd(['rev-parse', '--abbrev-ref', 'HEAD'], cwd).strip()
 
 
+def tag_exists(cwd, tag_name):
+    found_tag_name = run_git_cmd(['tag', '-l', tag_name],
+                                 cwd).strip()
+    return found_tag_name == tag_name
+
+
+def push_tag(cwd, tag_name, tag_message=None):
+    if tag_message is None:
+        tag_message = tag_name
+
+    run_git_cmd(['tag', '-a', tag_name, '-m', tag_message],
+                cwd).strip()
+    run_git_cmd(['push', 'origin', tag_name],
```

Comment:
Won't this push the commits as well? This gets called for the development deploys too as far as I can tell, which might contain unpublished code